devito/types/basic.py

```diff
@@ -773,7 +773,7 @@ def symbolic_shape(self):
                     for i, j, k in zip(domain, halo, padding))
         return DimensionTuple(*ret, getters=self.dimensions)
 
-    @property
+    @cached_property
```

Comment:
Why wasn't this cached previously?

Reply:
Not sure why, but don't see why it shouldn't be, probably just not caught